codegen/projections/high_score_service/sig/high_score_service/client.rbs

```diff
@@ -17,14 +17,14 @@ module HighScoreService
 
     attr_reader config: Config
 
-    def api_key_auth: (?::Hash[::Symbol, untyped] params, ?::Hash[::Symbol, untyped] options)-> Hearth::Output
-    def basic_auth: (?::Hash[::Symbol, untyped] params, ?::Hash[::Symbol, untyped] options)-> Hearth::Output
-    def bearer_auth: (?::Hash[::Symbol, untyped] params, ?::Hash[::Symbol, untyped] options)-> Hearth::Output
-    def create_high_score: (?::Hash[::Symbol, untyped] params, ?::Hash[::Symbol, untyped] options)-> Hearth::Output
-    def delete_high_score: (?::Hash[::Symbol, untyped] params, ?::Hash[::Symbol, untyped] options)-> Hearth::Output
-    def digest_auth: (?::Hash[::Symbol, untyped] params, ?::Hash[::Symbol, untyped] options)-> Hearth::Output
-    def get_high_score: (?::Hash[::Symbol, untyped] params, ?::Hash[::Symbol, untyped] options)-> Hearth::Output
-    def list_high_scores: (?::Hash[::Symbol, untyped] params, ?::Hash[::Symbol, untyped] options)-> Hearth::Output
-    def update_high_score: (?::Hash[::Symbol, untyped] params, ?::Hash[::Symbol, untyped] options)-> Hearth::Output
+    def api_key_auth: (?::Hash[::Symbol, untyped] params, ?::Hash[::Symbol, untyped] options)-> Hearth::Output[Types::ApiKeyAuthOutput]
```

Comment:
Is this correct? Output can have either an error or a type, under the accessor `data` or `error` - I think this should be as is?

Reply:
I think this is correct - Output is defined with a paramtraized type (`DATA`) which is the type of the `data` accessor - or more specially: `attr_accessor data: (DATA? | nil)` which is true.  The output's data method will always either return nil or the operation's output type.

Reply:
Hmm ok I couldn't remember if that change was still in the branch. Also I think DATA? Is sufficient because it's optional.

Reply:
I am thinking that parameterized is wrong though because we don't delegate to DATA like we do in v3. I think it should be reverted still. It depends on what the output looks like.

Reply:
I think parametrized is correct - delegation doesn't matter for that.    Hearth::Output is generic.  Each client method returns Hearth::Output with a concrete known type for the `data` accessor:

```
client.get_high_score(...).data&.high_score # we know that if data is not nil, then it has the high_score method
```